bok_choy/a11y/a11y_audit.py

```diff
@@ -0,0 +1,267 @@
+"""
+Interface for running accessibility audits on a PageObject.
+"""
+import json
+import os
+import requests
+
+from abc import abstractmethod, abstractproperty, ABCMeta
+from textwrap import dedent
+
+
+class AccessibilityError(Exception):
+    """
+    The page violates one or more accessibility rules.
+    """
+    pass
+
+
+class A11yAuditConfig(object):
+    """
+    The `A11yAuditConfig` object defines the options available in an
+    accessibility ruleset.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, *args, **kwargs):
+        super(A11yAuditConfig, self).__init__(*args, **kwargs)
+        self.rules_file = None
+
+    def set_rules_file(self, path=None):
+        """
+        Sets `self.rules_file` to the passed file.
+
+        Args:
+            A filepath where the JavaScript for the ruleset can be found.
+
+        This is intended to be used in the case of using an extended or
+        modified version of the ruleset. The interface and response
+        format are expected to be unmodified.
+        """
+        if path:
+            self.rules_file = os.path.abspath(path)
+
+    @abstractmethod
+    def set_rules(self, rules, ignore=False, include_type="rule"):
+        """
+        Overrides the default list of rules to be run.
+
+        Raises:
+
+            `NotImplementedError` if this isn't overwritten in the ruleset
+                specific implementation.
+        """
+        raise NotImplementedError(
+            "The ability to specify rules has not been implemented."
+        )
+
+    @abstractmethod
+    def set_scope(self, include=None, exclude=None):
+        """
+        Overrides the default scope (part of the DOM) to inspect.
+
+        Raises:
+
+            `NotImplementedError` if this isn't overwritten in the ruleset
+                specific implementation.
+        """
+        raise NotImplementedError(
+            "The ability to specify has not been implemented."
```

Comment:
nit: I think this needs a noun, such as "The ability to specify an area of the DOM..." or "the ability to specify where rules should be assessed..."